### DISCUSSION TITLE
ESP32 Berry Touch button

### DISCUSSION THREAD

**Discussion by martin072** [arendst/Tasmota#17229] [Q&A] [answered]
I've quick question for ESP32 and Berry scripting

I am looking to use a dimmer script that was written for the EPS8266 that utilises one button to turn on/off and dim a light.
Would that be able in Berry script as well? and more preferably, would you be able to access the (raw?) touch buttons on the ESP32 from Berry?

Thanks

**Comment by sfromis** [comment, marked as answer]
Yeah, using normal `tasmota.add_rule` you should be able to get actions on `Button1#state` and so on too. And you can configure ESP32 touch inputs as buttons too, `Button_tc`.

**Reply by martin072** [reply]
Thanks for the quick reply. I guess the rule-road will also interfere with the 'default' operation. i.e. touching it if it is configured it will switch on/off the channel? If to have full control in the script (i.e. let the script dim/on/off the channel) I guess rules would not work? Or does tasmota nowadays have this built in to use one button for multiple use? ie short tap: toggle power, long tap (and hold): dimmer increase to full and the decrease to low?

**Reply by sfromis** [reply]
>  I guess the rule-road will also interfere with the 'default' operation

No, rules in Berry will not supersede default actions, that is only with "classic" rules

While you have `SwitchMode 11` for dimmer mode, I'd not expect this to work with touch buttons.

**Reply by martin072** [reply]
Thanks

**Reply by martin072** [reply]
Got that working with the add_rule option, but not what I wanted ;) Is there also a way to read raw gpio data in Berry (like in scripting)?

**Reply by barbudor** [reply]
Yes, through the GPIO module which provide digitalRead()

**Reply by martin072** [reply]
Ok thanks. 

One thing i did try was switchmode 11 with the WS2812 (or with a PWM output for that matter, results were the same), but it doesn't seem to work on my ESP32-S2 (lolin s2). Other switchmodes work Ok, but 11 (or 12) does nothing. No dimming, no on/off toggle. whilst the mode 4 (for example) works fine. Anything I've missed here?
